src/ranged.cpp

```diff
@@ -258,25 +287,39 @@ bool player::handle_gun_damage( item &it, int shots_fired )
             }
         }
     }
-    if( curammo_effects.count( "BLACKPOWDER" ) ) {
-        if( !it.faults.count( fault_gun_blackpowder ) &&
-            it.faults_potential().count( fault_gun_blackpowder ) ) {
-            it.faults.insert( fault_gun_blackpowder );
-        }
-        if( one_in( firing->blackpowder_tolerance ) &&
-            it.faults_potential().count( fault_gun_clogged ) ) {
-            add_msg_player_or_npc( m_bad, _( "Your %s is clogged up with blackpowder fouling!" ),
-                                   _( "<npcname>'s %s is clogged up with blackpowder fouling!" ),
-                                   it.tname() );
-            it.faults.insert( fault_gun_clogged );
-            return false;
-        }
-        if( it.ammo_data()->ammo->recoil < firing->min_cycle_recoil && shots_fired > 0 ) {
-            add_msg_player_or_npc( m_bad, _( "Your %s fails to cycle!" ),
-                                   _( "<npcname>'s %s fails to cycle!" ),
-                                   it.tname() );
-            return false;
+    if( curammo_effects.count( "MUZZLE_SMOKE" ) || curammo_effects.count( "BLACKPOWDER" ) ) {
+        int blunderbussflintlock = 0;
+        int shot = 0;
+        if( it.dirt < 500 ) {
+            for( const ammotype &ammo : it.type->gun->ammo ) {
+                std::string compammo = ammo.c_str();
+                if( compammo == "shot" ) {
+                    shot = shot + 1;
+                } else if( compammo == "flintlock" || compammo == "blunderbuss" ) {
+                    blunderbussflintlock = blunderbussflintlock + 1;
+                }
+            }
+            if( blunderbussflintlock > 0 ) {
```

Comment:
I think all of this code would be much more simple if the number by which "dirt" variable is incremented was a JSON value for ammo (with default, for now, being 0, and setting it to other values for blackpowder cartridges).

Reply:
Yeah but then you'd need to modify every gun in the JSON along with all the ammo. I think it will be a lot of changes for very little benefit.

Reply:
Being able to fine-tune it for individual guns and to add blackpowder guns/ammo types via JSON only isn't "very little" benefit, I think.

Reply:
I think what I'll do is add some BP-related flags to guns. Like BP_RESIST to reduce fouling accumulation in that gun and BP_DESIGNED to reduce it even more.

Reply:
I was messing around with it and it looks like for now the other flags (manual, pump) are enough. More can easily be added in the future given the way I rewrote the code.